### PR TITLE
docs(operators): write comprehensive JSDoc on the buffer operators

### PR DESCRIPTION
**Pull request by staltz**
Merge after PR #1473 is merged

### CODE REVIEW

File: src/operator/switch.ts [hunk 1]
```diff
@@ -6,20 +6,43 @@ import {InnerSubscriber} from '../InnerSubscriber';
...
- * is emitted, switchOnNext stops emitting items from the earlier-emitted Observable and begins emitting items from the
- * new one.
+ * `switch` subscribes to an Observable that emits Observables,
+ * also known as a higher-order Observable. Each time it observes one of these
+ * emitted nested Observables, the output Observable begins emitting the items
+ * emitted by that nested Observable. So far, it behaves like {@link mergeAll}.
+ * However, when a new nested Observable is emitted, `switch` stops emitting
+ * items from the earlier-emitted nested Observable and begins emitting items
+ * from the new one. It continues to behave like this for subsequent nested
+ * Observables.
```
Comment (kwonoj) [inline src/operator/switch.ts]: Will it makes sense to mention `switch` also handles subscription when switching occurs? I remember someone have asked if it's needed to explicitly manage inner subscription after switching.

Comment (staltz) [inline src/operator/switch.ts]: Good point. But let's talk about this in PR https://github.com/ReactiveX/RxJS/pull/1473 which is the one that has the commit for switch.

Comment (kwonoj) [inline src/operator/switch.ts]: Cool :)

**Comment by kwonoj** [comment]
Merged with 56c9191, thanks @staltz

**Pull request closed without merging**